### PR TITLE
354 Define Specific SDLS EP SAs

### PR DESCRIPTION
**Pull request by rjbrown2**
Allow EP to only work with SA 0 or Max

### CODE REVIEW

File: test/unit/ut_ep_key_mgmt.c [hunk 1]
```diff
@@ -136,19 +114,6 @@ UTEST(EP_KEY_MGMT, ACTIVATE_141_142)
     char *buffer_nist_key_h = "000102030405060708090A0B0C0D0E0F000102030405060708090A0B0C0D0E0F";
     char *buffer_nist_iv_h  = "b6ac8e4963f49207ffd6374b"; // The last valid IV that was seen by the SA
     char *buffer_ACTIVATE_h = "2003001e00ff000000001880d038000c197f0b00020004008d008e82ebe4fc55555555";
-    //                        |2003001e00| = Primary Header
```
Comment (jlucas9) [inline test/unit/ut_ep_key_mgmt.c:139]: Are these breakdowns still useful? Seen in multiple places removed

File: test/unit/ut_ep_key_mgmt.c [hunk 2]
```diff
@@ -265,6 +219,7 @@ UTEST(EP_KEY_MGMT, DEACTIVATE_142)
     test_association->shsnf_len = 2;
     test_association->arsn_len  = 2;
     test_association->arsnw     = 5;
+    test_association->sa_state = SA_OPERATIONAL;
```
Comment (jlucas9) [inline test/unit/ut_ep_key_mgmt.c:222]: Looks like this was already set above, can likely remove that one

**Review by dccutrig — APPROVED** [review]

**Review by jlucas9 — APPROVED** [review]
Only minor things commented. Looks good

**Review by Donnie-Ice — APPROVED** [review]

**Review by jlucas9 — APPROVED** [review]

File: src/core/crypto_tc.c [hunk 3]
```diff
@@ -1924,10 +1925,20 @@ int32_t Crypto_TC_ProcessSecurity_Cam(uint8_t *ingest, int *len_ingest, TC_t *tc
         return status; // Cryptography IF call failed, return.
     }
     // Extended PDU processing, if applicable
-    // TODO: Validiate using correct SA
+ 
     if (status == CRYPTO_LIB_SUCCESS && crypto_config.process_sdls_pdus == TC_PROCESS_SDLS_PDUS_TRUE)
     {
-        status = Crypto_Process_Extended_Procedure_Pdu(tc_sdls_processed_frame, ingest);
+        if((sa_ptr->spi == SPI_MIN) || sa_ptr->spi == SPI_MAX) //TODO:  Make sure that SPI can't be changed from the current array index.
```
Comment (dccutrig) [inline src/core/crypto_tc.c:1931]: Line 1931: Are we better off to #DEFINE NUM_SA = 64, then in the check assert that sa_ptr->spi >0 and <NUM_SA? Currently we have an off by one case where I feel like we've unintentionally defined 65 SAs here (0 to 64 inclusive). Besides that this logic makes sense.

**Review by dccutrig — COMMENTED** [review]
Comment left crypto_tc line 1931

**Pull request merged**